### ISSUE TITLE
Under what circumstances would I want to attach my context to the InheritedWidget

### ISSUE THREAD

**Issue by larryq** [robertohuertasm/generic_bloc_provider#1]
My apologies, but opening an issue appears to be the only way I can ask a question.  Your library is much appreciated but I was wondering when I might want to attach my context to the InheritedWidget vs not.  Are there advantages to using inheritFromWidgetOfExactType vs  ancestorInheritedElementForWidgetOfExactType under certain circumstances?

**Comment by robertohuertasm** [comment]
Hi Larry, I saw your message in the blog but I'm a little bit busier than usual these days. Anyway, I think it's better to ask here so everyone can participate and suggest changes or improvements.

Regarding your question, if you refer to the Flutter documentation about the [InheredtedWidget](https://docs.flutter.io/flutter/widgets/InheritedWidget-class.html) you can see the following:

> To obtain the nearest instance of a particular type of inherited widget from a build context, use BuildContext.inheritFromWidgetOfExactType. 

> Inherited widgets, when referenced in this way, will cause the consumer to rebuild when the inherited widget itself changes state.

Pay special attention to the second paragraph. 

Also, from the [inheritFromWidgetOfExactType](https://docs.flutter.io/flutter/widgets/BuildContext/inheritFromWidgetOfExactType.html) docs:

> Obtains the nearest widget of the given type, which must be the type of a concrete InheritedWidget subclass, and registers this build context with that widget such that when that widget changes (or a new widget of that type is introduced, or the widget goes away), this build context is rebuilt so that it can obtain new values from that widget.

So basically, if you use `inheritFromWidgetOfExactType` (the default in this library) any change in the state of the `BlocProvider`, which is an `InheritedWidget`, will cause the build context used to retrieve it to be rebuilt. 

If you use [ancestorInheritedElementForWidgetOfExactType](https://docs.flutter.io/flutter/widgets/BuildContext/ancestorInheritedElementForWidgetOfExactType.html) this won't happen. It's that simple.

So if you want your widgets to get rebuilt once the `BlocProvider` state changes (not that this has nothing to do with the streams that your `Bloc` may have), or a new instance is introduced or goes away, it's the way to go.

**Comment by larryq** [comment]
I see now, thanks for the clarification.  The flutter docs, which are very good, were a bit confusing to me in this instance; for example, when saying: 

> this build context with that widget

 I wasn't quite sure what **this build context** and **that widget**  were referring to.  Was 'that widget' the InheritedWidget or my own, for example?  I suspected the former but really couldn't tell for sure.  Appreciate the assist!
